cmd/esc/cli/env_version.go

```diff
@@ -3,27 +3,126 @@
+	var utc bool
+
 	cmd := &cobra.Command{
-		Use:   "version",
-		Short: "Manage version tags",
-		Long: "Manage version tags\n" +
+		Use:   "version [<org-name>/]<environment-name>@<version>",
+		Args:  cobra.ExactArgs(1),
+		Short: "Manage versions",
+		Long: "Manage versions\n" +
+			"\n" +
+			"This command describes the referenced environment version.\n" +
 			"\n" +
-			"This command creates, inspects, updates, deletes, and lists version tags.\n" +
-			"A version tag is a name that refers to a specific revision of an environment.\n" +
-			"Once created, version tags can be updated to refer to new reversions\n" +
-			"of an environment. Version tags can be used to refer to a particular logical\n" +
-			"version of an environment rather than a specific revision.\n",
+			"Subcommands exist for viewing revision history and managing" +
+			"tagged versions.",
 		SilenceUsage: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			if err := env.esc.getCachedClient(ctx); err != nil {
+				return err
+			}
+
+			orgName, envName, version, args, err := env.getEnvName(args)
+			if err != nil {
+				return err
+			}
+			if version == "" {
+				return errors.New("please specify a version")
+			}
+			_ = args
+
+			st := style.NewStylist(style.Profile(env.esc.stdout))
+			if isRevisionNumber(version) {
+				revisionNumber, err := strconv.ParseInt(version, 10, 0)
+				if err != nil {
+					return err
+				}
+				rev, err := env.esc.client.GetEnvironmentRevision(ctx, orgName, envName, int(revisionNumber))
+				if err != nil {
+					return err
+				}
+				printRevision(env.esc.stdout, st, *rev, utc)
+			} else {
+				tag, err := env.esc.client.GetEnvironmentRevisionTag(ctx, orgName, envName, version)
+				if err != nil {
+					return err
+				}
+				printRevisionTag(env.esc.stdout, st, *tag, utc)
+			}
+			return nil
+		},
 	}
 
-	cmd.AddCommand(newEnvVersionShowCmd(env))
 	cmd.AddCommand(newEnvVersionTagCmd(env))
-	cmd.AddCommand(newEnvVersionRmCmd(env))
-	cmd.AddCommand(newEnvVersionLsCmd(env))
+	cmd.AddCommand(newEnvVersionHistoryCmd(env))
+
+	cmd.Flags().BoolVar(&utc, "utc", false, "display times in UTC")
 
 	return cmd
 }
+
+func printRevision(stdout io.Writer, st *style.Stylist, r client.EnvironmentRevision, utc bool) {
+	rules := style.Default()
+
+	st.Fprintf(stdout, rules.H1.StylePrimitive, "revision %v", r.Number)
+	switch len(r.Tags) {
+	case 0:
+		// OK
+	case 1:
+		st.Fprintf(stdout, rules.LinkText, " (tag: %v)", r.Tags[0])
+	default:
+		st.Fprintf(stdout, rules.LinkText, " (tags: %v)", strings.Join(r.Tags, ", "))
+	}
+	fmt.Fprintln(stdout, "")
+
+	if r.CreatorLogin == "" {
+		fmt.Fprintf(stdout, "Author: <unknown>\n")
+	} else {
+		fmt.Fprintf(stdout, "Author: %v <%v>\n", r.CreatorName, r.CreatorLogin)
+	}
+
+	stamp := r.Created
+	if utc {
+		stamp = stamp.UTC()
+	} else {
+		stamp = stamp.Local()
+	}
+
+	fmt.Fprintf(stdout, "Date:   %v\n", stamp)
```

Comment:
Sorry if I am jumping the gun but the changelog said final.  We talked about outputting the tag metadata as well if the version passed in was a tag.